pkg/controller/scylladbcluster/resource.go

```diff
@@ -515,22 +523,40 @@ func calculateEndpointsForRemoteDCPods(sc *scyllav1alpha1.ScyllaDBCluster, broad
 	case scyllav1alpha1.BroadcastAddressTypeServiceClusterIP:
 		eps, err := makeRemoteServiceEndpoints(sc, remoteDC, remoteDCNamespace, remoteDCPodSelector, remoteServiceLister, makeServiceClusterIPEndpoints)
 		if err != nil {
-			return nil, fmt.Errorf("can't make remote service endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), remoteDC.Name, err)
+			return nil, nil, fmt.Errorf("can't make remote service endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), remoteDC.Name, err)
 		}
 		endpoints = append(endpoints, eps...)
 
 	case scyllav1alpha1.BroadcastAddressTypeServiceLoadBalancerIngress:
 		eps, err := makeRemoteServiceEndpoints(sc, remoteDC, remoteDCNamespace, remoteDCPodSelector, remoteServiceLister, makeServiceLoadBalancerIngressEndpoints)
 		if err != nil {
-			return nil, fmt.Errorf("can't make remote service endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), remoteDC.Name, err)
+			return nil, nil, fmt.Errorf("can't make remote service endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), remoteDC.Name, err)
 		}
 		endpoints = append(endpoints, eps...)
 
 	default:
-		return nil, fmt.Errorf("unsupported node broadcast address type %v specified in %q ScyllaDBCluster", broadcastAddressType, naming.ObjRef(sc))
+		return nil, nil, fmt.Errorf("unsupported node broadcast address type %v specified in %q ScyllaDBCluster", broadcastAddressType, naming.ObjRef(sc))
+	}
+
+	// Detect IP family from the first endpoint address
+	for _, endpoint := range endpoints {
+		if len(endpoint.Addresses) > 0 {
+			parsedIP := net.ParseIP(endpoint.Addresses[0])
+			if parsedIP != nil {
+				if helpers.IsIPv6(parsedIP) {
+					ipFamily := corev1.IPv6Protocol
+					detectedIPFamily = &ipFamily
+					break
+				} else if helpers.IsIPv4(parsedIP) {
+					ipFamily := corev1.IPv4Protocol
+					detectedIPFamily = &ipFamily
+					break
+				}
+			}
+		}
 	}
 
```

Comment:
I believe this deserves a separate helper function with its own set of unit tests, WDYT?